src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/updaters/PtvServiceEntityUpdater.java

```diff
@@ -158,11 +167,25 @@ private void updatePtvService(ServiceId ptvServiceId, Long orderIndex) {
         
         index(identifier.getKuntaApiId(), service, orderIndex);
       }
+      
+      List<PageId> parentPageIds = identifierRelationController.listPageIdsByChildId(ptvServiceId);
+      updateParentPageIds(parentPageIds);
     } else {
       logger.warning(String.format("Service %s processing failed on [%d] %s", ptvServiceId.getId(), response.getStatus(), response.getMessage()));
     }
   }
 
+  private void updateParentPageIds(List<PageId> parentPageIds) {
+    for (PageId parentPageId : parentPageIds) {
+      Identifier parentPageIdentifier = identifierController.findIdentifierById(parentPageId);
+      if (ManagementConsts.IDENTIFIER_NAME.equals(parentPageIdentifier.getSource())) {
+        // TODO: REFACTOR THIS AWAY FROM THE PTV INTEGRATION PLUGIN
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)